common/analysis/lint_waiver_config.lex

```diff
@@ -0,0 +1,98 @@
+%{
+#define _WAIVER_FLEXLEXER_H_
+#include <iostream>
+#include "common/analysis/lint_waiver_config.h"
+%}
+
+%option nodefault
+%option noyywrap
+%option prefix="verible"
+%option c++
+%option yyclass="verible::LintWaiverConfig"
+
+/* white space */
+LineTerminator \r|\n|\r\n
+InputCharacter [^\r\n\0]
+Space [ \t\f\b]
+
+DecimalDigits [0-9]+
+
+Alpha [a-zA-Z]
+RuleStart {Alpha}
+Letter {RuleStart}|"$"
+
+RuleName {RuleStart}({Letter}|-)*
+
+
+RuleParam -rule
+LineParam -line
```

Comment:
Would it make sense to generalize this to something `Flag -{RuleName}` ?

Reply:
Changed the lexer to accept arbitrary flags